perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -721,6 +721,8 @@ public void setAttributes(PerunSession sess, Facility facility, List<Attribute>
 				}
 			}
 		}
+		//if checkAttributesSyntax fails it causes rollback so no attribute will be stored
+		checkAttributesSyntax(sess, facility, attributesToSet);
```

Comment:
As @balcirakpeter noted, you can perform check on syntax before actually storing attributes to the DB (before call ``setAttributeWithoutCheck()``).  I believe this applies to all changed methods.